composer/trainer/trainer.py

```diff
@@ -179,10 +179,12 @@ def __init__(
         self.device = device
 
         if not seed:
-            # Set a deterministic seed in the hparams
-            # This seed will be dumped in the hparams that are saved with checkpoints
             seed = get_random_seed()
             log.info(f"Seed was None. Setting seed to random value: {seed}")
+
+        # Assure that each process has a different seed, necessary if a seed is passed to init
+        seed += ddp.get_global_rank()
```

Comment:
When resuming from a checkpoint, would the global rank effectively be added twice (once upon the first training run, and again when resuming?)

Reply:
Nope because when restoring from checkpoint the seed is loaded from the save file in the `load_checkpoint` function and then `seed_all` is called with that seed so this codepath is nullified.